src/optimizer/expr_simplification.ts

```diff
@@ -0,0 +1,713 @@
+                ctx: ctx,
+            };
+        }
+        case "statement_repeat": {
+            const expressionResult = process_expression(
+                stmt.iterations,
+                ctx,
+                interpreter,
+            );
+            const newIterations = expressionResult.expr;
+            ctx = expressionResult.ctx;
+
+            const statementsResult = process_statements(
+                stmt.statements,
+                ctx,
+                interpreter,
+            );
+            const newStatements = statementsResult.stmts;
+            ctx = statementsResult.ctx;
+
+            // Create the replacement node
+            return {
+                stmt: cloneAstNode({
+                    ...stmt,
+                    iterations: newIterations,
+                    statements: newStatements,
+                }),
+                ctx: ctx,
+            };
+        }
+        case "statement_try": {
+            const statementsResult = process_statements(
+                stmt.statements,
+                ctx,
+                interpreter,
+            );
+            const newStatements = statementsResult.stmts;
+            ctx = statementsResult.ctx;
+
+            // Create the replacement node
+            return {
+                stmt: cloneAstNode({
+                    ...stmt,
+                    statements: newStatements,
+                }),
+                ctx: ctx,
+            };
+        }
+        case "statement_try_catch": {
+            const statementsResult = process_statements(
+                stmt.statements,
+                ctx,
+                interpreter,
+            );
+            const newStatements = statementsResult.stmts;
+            ctx = statementsResult.ctx;
+
+            const catchStatementsResult = process_statements(
+                stmt.catchStatements,
+                ctx,
+                interpreter,
+            );
+            const newCatchStatements = catchStatementsResult.stmts;
+            ctx = catchStatementsResult.ctx;
+
+            // Create the replacement node
+            return {
+                stmt: cloneAstNode({
+                    ...stmt,
+                    statements: newStatements,
+                    catchStatements: newCatchStatements,
+                }),
+                ctx: ctx,
+            };
+        }
+    }
+}
+
+function process_expression(
+    expr: AstExpression,
+    ctx: CompilerContext,
+    interpreter: Interpreter,
+): { expr: AstExpression; ctx: CompilerContext } {
+    const value = tryExpressionSimplification(expr, interpreter);
+    let newExpr = expr;
+    if (value !== undefined) {
+        try {
+            newExpr = makeValueExpression(value, expr.loc);
+            // Register the new expression in the context
+            ctx = registerAllSubExpTypes(ctx, newExpr, getExpType(ctx, expr));
+        } catch (_) {
+            // This means that transforming the value into an AST node is
```

Comment:
We could have any kind of exception here, including stack overflow exception.

If there is a specific kind of exception that should be handled here, there has to be an `instanceof` check, and all other exceptions have to be rethrown up the stack.